packages/core/src/prompts/prompt.ts

```diff
@@ -168,7 +173,17 @@ export default class Prompt {
 
 		if (key?.name === 'return') {
 			if (this.opts.validate) {
-				const problem = this.opts.validate(this.value);
+				this.state = 'validate';
+				let problem = this.opts.validate(this.value);
+				// Only trigger validation state after 300ms.
+				// If problem resolves first, render will be cancelled.
+				await Promise.race([
+					problem,
+					setTimeout(300).then(() => {
```

Comment:
Magic number? Constant with descriptive name?

Just curious, but isn't 400 the typical breakpoint in these scenarios?

Reply:
Yeah, magic number. Moving to a constant with descriptive name is probably better! 400 works for me 👍🏻